File: Data Structures & Algorithms/valid-binary-search-tree/submission-7.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private List<Integer> res = new ArrayList<>();
    public boolean isValidBST(TreeNode root) {
        dfs(root);
        for(int i = 1; i < res.size(); i++){
            if(res.get(i-1) >= res.get(i)){
                return false;
            }
        }
        return true;
    }
    private void dfs(TreeNode root){
        if(root == null) return;
        dfs(root.left);
        res.add(root.val);
        dfs(root.right);
    }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-7.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        isBalance = True
        def checkBalance(root):
            nonlocal isBalance
            if not root:
                return -1
            left = checkBalance(root.left)
            right = checkBalance(root.right)
            if(abs(left-right)>1):
                isBalance=False
            return 1 + max(left, right)
        checkBalance(root)
        return isBalance
        